etcdutl/etcdutl/check_command.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdutl
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+
+	"go.etcd.io/etcd/pkg/v3/cobrautl"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/snap"
+	"go.etcd.io/etcd/server/v3/storage/backend"
+	"go.etcd.io/etcd/server/v3/storage/datadir"
+	"go.etcd.io/etcd/server/v3/storage/schema"
+	"go.etcd.io/etcd/server/v3/storage/wal"
+
+	"github.com/spf13/cobra"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	checkDataDir           string
+	ConsistentIndexKeyName = []byte("consistent_index")
+)
+
+// NewCheckCommand returns the cobra command for "check".
+func NewCheckCommand() *cobra.Command {
+	cc := &cobra.Command{
+		Use:   "check <subcommand>",
+		Short: "commands for checking properties of the etcd cluster",
+	}
+
+	cc.AddCommand(NewCheckBrokenCommand())
+
+	return cc
+}
+
+// NewCheckBrokenCommand returns the cobra command for "check broken".
+func NewCheckBrokenCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "broken [options]",
+		Short: "Check whether the data file is broken",
+		Run:   newCheckBrokenCommand,
+	}
+
+	cmd.Flags().StringVar(&checkDataDir, "data-dir", "", "Path to the output data directory")
+	cmd.MarkFlagRequired("data-dir")
+
+	return cmd
+}
+
+func newCheckBrokenCommand(cmd *cobra.Command, args []string) {
+	walSnaps, err := wal.ValidSnapshotEntries(GetLogger(), datadir.ToWalDir(checkDataDir))
```

Comment:
nit: I don't think this will help with `panic: tocommit(1808562) is out of range [1808561]` error that you've mentioned in the issue [comment](https://github.com/etcd-io/etcd/issues/14837#issuecomment-1325906170) Here is longer [discussion](https://github.com/etcd-io/etcd/issues/10166) why that error happens.